File: go/problem#1922.go
package main

func countGoodNumbers(n int64) int {
	var mod int64 = 1e9 + 7
	var even int64 = (n + 1) / 2
	var odd int64 = n / 2
	var pow func(x, y int64) int64
	pow = func(x, y int64) int64 {
		if y == 0 {
			return 1
		}
		if y%2 == 1 {
			return (x * pow(x, y-1)) % mod
		}
		xx := (x * x) % mod
		y = y / 2
		return pow(xx, y)
	}
	ret := (pow(5, even) * pow(4, odd)) % mod
	return int(ret)
}

File: c/problems/problem#1922.c
#define ull unsigned long long

ull mod = 1e9 + 7;

ull pow1922(ull x, ull y) {
  if (y == 0) {
    return 1;
  }
  if (y % 2 == 1) {
    return (x * pow1922(x, y - 1)) % mod;
  }
  ull xx = (x * x) % mod;
  y = y / 2;
  return pow1922(xx, y);
}

int countGoodNumbers(ull n) {
  ull even = (n + 1) / 2;
  ull odd = n / 2;
  ull ans = (pow1922(5, even) * pow1922(4, odd)) % mod;
  return ans;
}
